topology/topology.go

```diff
@@ -72,50 +74,60 @@ type TopologyProvider struct {
 	fetcher   Fetcher
 }
 
-func (t *TopologyProvider) NetworkTopology(hash string) (NetworkTopology, error) {
+func (t *TopologyProvider) NetworkTopology(hash string) (*NetworkTopology, error) {
+	if hash != "" {
+		log.Info().Msgf("New NetworkTopology initialisation. Hash: %s, url: %s", hash, t.url)
+	}
 	resp, err := t.fetcher.Get(t.url)
 	if err != nil {
-		return NetworkTopology{}, err
+		return nil, err
 	}
-
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return NetworkTopology{}, err
+		return nil, err
+	}
+	response := strings.TrimSuffix(string(body), "\n")
+	ct, err := hex.DecodeString(response)
+	if err != nil {
+		return nil, err
 	}
-
 	h := sha256.New()
-	h.Write(body)
+	h.Write(ct)
 	eh := hex.EncodeToString(h.Sum(nil))
 	if hash != "" && eh != hash {
-		return NetworkTopology{}, fmt.Errorf("topology hash %s not matching expected hash %s", string(eh), hash)
+		return nil, fmt.Errorf("topology hash %s not matching expected hash %s", string(eh), hash)
 	}
 
-	unecryptedBody := t.decrypter.Decrypt(string(body))
+	unecryptedBody := t.decrypter.Decrypt(ct)
 	rawTopology := &RawTopology{}
 	err = json.Unmarshal(unecryptedBody, rawTopology)
 	if err != nil {
-		return NetworkTopology{}, err
+		return nil, err
+	}
+	if hash != "" {
+		log.Info().Msgf("New NetworkTopology initialised. "+
```

Comment:
I would move this to app.go where it is intialized. Doesn't make sense to me to have if logic here to log something.